script.js

```diff
@@ -0,0 +1,141 @@
+// Otherwise...
+// When 'a' is less than b, return -1.
+// When 'a' is greater than b, return 1.
+// If margin is not given, treat it as if it were zero.
+// Assume: margin >= 0
+// Tip: Some languages have a way to make parameters optional.
+
+function isTheFirstNumberCloseToTheSecondNumber(firstNumber, secondNumber, marginParameter = 0) {
+    const absoluteValueOfNumbers = Math.abs(firstNumber - secondNumber);
+    if (absoluteValueOfNumbers <= marginParameter) {
+        return 0;
+    }
+    if (firstNumber < secondNumber) {
+        return -1;
+    }
+    if (firstNumber > secondNumber) {
+        return 1;
+    }
+}
+
+console.log(isTheFirstNumberCloseToTheSecondNumber(9,5, 0));
+
+// 3.You need to implement two functions, xor and or, that replicate the behaviour of their respective operators:
+// xor = Takes 2 values and returns true if, and only if, one of them is truthy.
+// or = Takes 2 values and returns true if either one of them is truthy.
+// When doing so, you cannot use the or operator: ||.
+
+// Not all input will be booleans - there will be truthy and falsey values [the latter including also empty strings and empty arrays]
+// There will always be 2 values provided.
+
+function isOnlyOneOfThemTruthy(firstValue, secondValue) {
+    if (firstValue && !secondValue) {
+        return true;
+    }
+    return false;
+}
+
+function isEitherOneOfThemTruthy(firstValue, secondValue) {
+    if (!firstValue && !secondValue) {
+        return false;
+    }
+    return true;
+}
+
+console.log(isOnlyOneOfThemTruthy("Bye"));
+console.log(isEitherOneOfThemTruthy(0, 2));
+
+// 4.Create a function called _if which takes 3 arguments: a value bool and 2 functions (which do not take any parameters): func1 and func2
+// When bool is truthy, func1 should be called, otherwise call the func2.
+
+function firstFunction() {
+    console.log('True')}
+function secondFunction() {
+    console.log('False')}
+
+function isBoolTruthy(boolValue, firstFunction, secondFunction) {
+    if (boolValue) {
+        return firstFunction();
+    }
+    return secondFunction();
+}
+
+isBoolTruthy(true, firstFunction, secondFunction);
+
+// 5.After a hard quarter in the office you decide to get some rest on a vacation.
+// So you will book a flight for you and your girlfriend and try to leave all the mess behind you.
+// You will need a rental car in order for you to get around in your vacation.
+// The manager of the car rental makes you some good offers.
+// Every day you rent the car costs $40. If you rent the car for 7 or more days, you get $50 off your total.
+// Alternatively, if you rent the car for 3 or more days, you get $20 off your total.
+// Write a code that gives out the total amount for different days(d).
+
+function getTheTotalAmountForCarRental(dayNumber) {
+    const costPerDay = 40;
+    const fiftyOff = 50;
+    const twentyOff = 20;
+    if (dayNumber >= 7) {
+        return costPerDay * dayNumber - fiftyOff;
```

Comment:
You are using `costPerDay * dayNumber` three times in this function - keep it in a separate variable at the top of this function.